ui/src/app/utils/index.tsx

```diff
@@ -463,3 +463,32 @@ export function highlightSearchTerm(searchTerm: string, stringToHighlight: strin
       {word}
     </span>);
 }
+
+/*
+ * A method to run an api call with a specified number of retries and exponential backoff.
+ * This method will only error
+ * Parameters:
+ *    apiCall: Lambda that will run an API call, in the form of () => apiClient.apiCall(args)
+ *    maxRetries: The amount of retries the system will take before erroring
+ *    defaultWaitTime: How long the base exponential backoff is, in milliseconds
+ *      For example, if 1000 is passed in, it will wait 1s for the first retry,
+ *      2s for the second, etc.
+ */
+export async function apiCallWithGatewayTimeoutRetries<T>(
+  apiCall: () => Promise<T>, maxRetries = 3, defaultWaitTime = 1000): Promise<T> {
```

Comment:
Sorry for another naming nit: this isn't technically a "default" but rather an "initial" value. Maybe "initialWaitTime" or "startingWaitTime"